### PR TITLE
multipart upload implementation

### PR DESCRIPTION
**Pull request by art-vasilyev**
* support multipart upload from file resource
* support batch files direct uploading
* support upload from url in File.upload method

## Checklist

- [x] Tests (if applicable)
- [x] Documentation (if applicable)
- [x] Changelog stub (or use [conventional commit messages](https://www.conventionalcommits.org/))

### CODE REVIEW

File: pyuploadcare/resources/file.py [hunk 1]
```diff
@@ -49,8 +58,15 @@ class File(ApiMixin):
...
     """
 
+    # batch size for multiple delete and store requests
     batch_chunk_size = 500
 
+    #  minimum file size for multipart uploads
+    multipart_min_file_size = 10485760
+
+    # chunk size for multipart uploads
+    multipart_chunk_size = 5 * 1024 * 1024
```
Comment (dmitry-mukhin) [inline pyuploadcare/resources/file.py:68]: ideally, this should be configurable

File: pyuploadcare/resources/file.py [hunk 2]
```diff
@@ -328,12 +344,12 @@ def create_remote_copy(
         response = self.files_api.remote_copy(**data)
         return response.result
 
-    def delete(self):
+    def delete(self) -> "None":
```
Comment (dmitry-mukhin) [inline pyuploadcare/resources/file.py:347]:
```suggestion
    def delete(self) -> None:
```

File: pyuploadcare/resources/file.py [hunk 3]
```diff
@@ -353,45 +369,229 @@ def construct_from(cls, file_info):
...
+    @staticmethod
+    def _get_file_size(file_object: IO) -> int:
+        return os.fstat(file_object.fileno()).st_size
+
     @classmethod
-    def upload(
+    def upload(  # noqa: C901
         cls,
-        file_obj,
+        file_obj_or_url: Union[IO, str],
```
Comment (dmitry-mukhin) [inline pyuploadcare/resources/file.py:379]:
```suggestion
        file_handle: Union[IO, str],
```
is it a better name?

File: pyuploadcare/resources/file.py [hunk 4]
```diff
@@ -353,45 +369,229 @@ def construct_from(cls, file_info):
...
+        if size is None:
+            size = cls._get_file_size(file_obj)
+
+        if not mime_type:
+            mime_type, _encoding = mimetypes.guess_type(file_obj.name)
+
+        if not mime_type:
+            raise ValueError(
+                "Unable to determine file mime type, please set manually"
+            )
```
Comment (dmitry-mukhin) [inline pyuploadcare/resources/file.py:561]: why should we fail such uploads?
i believe setting a default mime (`application/octet-stream`) is better than failing

File: pyuploadcare/resources/file.py [hunk 5]
```diff
@@ -353,45 +369,229 @@ def construct_from(cls, file_info):
...
+
+        complete_response = cls.upload_api.start_multipart_upload(
```
Comment (dmitry-mukhin) [inline pyuploadcare/resources/file.py:563]:
```suggestion
        start_response = cls.upload_api.start_multipart_upload(
```

File: README.rst [hunk 6]
```diff
@@ -104,6 +104,66 @@ or, if you prefer it the old way:
...
+Upload multiple files. Direct upload method is used::
+
+    file1 = open('file1.txt')
+    file2 = open('file2.txt')
+    ucare_files: List[File] = File.upload_files([file1, file2])
+
+Send single file via multipart upload::
+
+    with open('file.txt') as file_object:
+        ucare_file: File = File.multipart_upload(file_object)
```
Comment (dmitry-mukhin) [inline README.rst:142]:
```suggestion
        ucare_file: File = File.upload(file_object)
```

**Review by dmitry-mukhin — CHANGES_REQUESTED** [review]

File: pyuploadcare/resources/file.py (hunk 2 above)
Comment (art-vasilyev) [reply pyuploadcare/resources/file.py]: done

File: pyuploadcare/resources/file.py (hunk 3 above)
Comment (art-vasilyev) [reply pyuploadcare/resources/file.py]: done

File: pyuploadcare/resources/file.py (hunk 4 above)
Comment (art-vasilyev) [reply pyuploadcare/resources/file.py]: done

File: pyuploadcare/resources/file.py (hunk 5 above)
Comment (art-vasilyev) [reply pyuploadcare/resources/file.py]: done

File: README.rst (hunk 6 above)
Comment (art-vasilyev) [reply README.rst]: done

File: pyuploadcare/resources/file.py (hunk 1 above)
Comment (dmitry-mukhin) [reply pyuploadcare/resources/file.py:66]: it's not a deal breaker, just noticing.

Comment (dmitry-mukhin) [reply pyuploadcare/resources/file.py:66]: but, please let it be configurable via config files, not only in client init

**Review by dmitry-mukhin — APPROVED** [review]

**Pull request merged**